neard/tests/node_cluster.rs

```diff
@@ -0,0 +1,104 @@
+use futures::future;
+
+use near_actix_test_utils::{run_actix_until_stop, spawn_interruptible};
+use near_client::{ClientActor, ViewClientActor};
+use near_primitives::types::{BlockHeight, BlockHeightDelta, NumSeats, NumShards};
+use testlib::{start_nodes, test_helpers::heavy_test};
+
+#[derive(Debug, Default)]
+pub struct NodeCluster {
+    dirs: Vec<tempfile::TempDir>,
+    is_heavy: bool,
+    num_shards: Option<NumShards>,
+    num_validator_seats: Option<NumSeats>,
+    num_lightclient: Option<usize>,
+    epoch_length: Option<BlockHeightDelta>,
+    genesis_height: Option<BlockHeight>,
+}
+
+macro_rules! add_mut_helpers {
+    ($($method:ident($self:ident $(,$arg:ident:$type:ty)?) => $expr:expr),+) => {
+        $(
+            pub fn $method(mut $self$(, $arg: $type)?) -> Self {
+                $expr;
+                $self
+            }
+        )+
+    }
+}
+
+impl NodeCluster {
+    pub fn new<F: Fn(usize) -> String>(capacity: usize, gen_dirname: F) -> Self {
+        Self {
+            dirs: (0..capacity)
+                .map(|index| {
+                    tempfile::Builder::new().prefix(&gen_dirname(index)).tempdir().unwrap()
+                })
+                .collect(),
+            ..Default::default()
+        }
+    }
+
+    add_mut_helpers! {
```

Comment:
We tend to avoid macroses if possible since it creates burden of “learning” the macro implementation. In this case, the functions are small, and I believe that even the total lines count could be the same without the macro while the complexity will be much lower, so I argue that it does not buy us anything here. Also, we cannot properly document the methods

Reply:
On documentation, I figured it was an internal implementation detail and with reasonable names, we wouldn't really need those, but I get it

> and I believe that even the total lines count could be the same

you're ~right, only +10 LoC